modules/monitoring-resources-reference-for-the-cluster-monitoring-operator.adoc

```diff
@@ -0,0 +1,155 @@
+// source code for the Cluster Monitoring Operator. Any changes made to this
+// file will be overwritten when the content is regenerated. If you wish to
+// make edits or learn more about how this file is generated, read the docgen utility
+// instructions in the source code for the CMO.
+:_mod-docs-content-type: REFERENCE
+[id="resources-reference-for-the-cluster-monitoring-operator"]
+= Resources reference for the Cluster Monitoring Operator
+
+This document describes the following resources deployed and managed by the Cluster Monitoring Operator (CMO):
+
+* link:#cmo-routes-resources[Routes]
+* link:#cmo-services-resources[Services]
+
+Use this information when you want to configure API endpoint connections to retrieve, send, or query metrics data.
+
+[IMPORTANT]
+====
+In certain situations, accessing API endpoints can degrade the performance and scalability of your cluster, especially if you use endpoints to retrieve, send, or query large amounts of metrics data.
+
+To avoid these issues, follow these recommendations:
+
+* Avoid querying endpoints frequently. Limit queries to a maximum of one every 30 seconds.
+* Do not try to retrieve all metrics data via the `/federate` endpoint for Prometheus.
+Query it only when you want to retrieve a limited, aggregated data set.
+For example, retrieving fewer than 1,000 samples for each request helps minimize the risk of performance degradation.
+====
+
+[id="cmo-routes-resources"]
+== CMO routes resources
+
+=== openshift-monitoring/alertmanager-main
+
+Expose the `/api` endpoints of the `alertmanager-main` service via a router.
+
+=== openshift-monitoring/prometheus-k8s
+
+Expose the `/api` endpoints of the `prometheus-k8s` service via a router.
+
+=== openshift-monitoring/prometheus-k8s-federate
+
+Expose the `/federate` endpoint of the `prometheus-k8s` service via a router.
+
+=== openshift-user-workload-monitoring/federate
+
+Expose the `/federate` endpoint of the `prometheus-user-workload` service via a router.
+
+=== openshift-monitoring/thanos-querier
+
+Expose the `/api` endpoints of the `thanos-querier` service via a router.
+
+=== openshift-user-workload-monitoring/thanos-ruler
+
+Expose the `/api` endpoints of the `thanos-ruler` service via a router.
+
+[id="cmo-services-resources"]
+== CMO services resources
+
+=== openshift-monitoring/prometheus-operator-admission-webhook
+
+Expose the admission webhook service which validates `PrometheusRules` and `AlertmanagerConfig` custom resources on port 8443.
+
+=== openshift-user-workload-monitoring/alertmanager-user-workload
+
+Expose the user-defined Alertmanager web server within the cluster on the following ports:
+
+* Port 9095 provides access to the Alertmanager endpoints. Granting access requires binding a user to the `monitoring-alertmanager-api-reader` role (for read-only operations) or `monitoritoring-alertmanager-api-writer` role in the `openshift-user-workload-monitoring` project.
```

Comment:
I think there is a typo here: "`monitoritoring-alertmanager-api-writer`"

Reply:
Good catch! I'll go ahead and fix it here, and write up a ticket to fix it in the source, too. Thanks!